eng/pipelines/corefx-base.yml

```diff
@@ -136,26 +136,25 @@ jobs:
         steps:
           - ${{ if ne(job.preBuildSteps[0], '') }}:
             - ${{ job.preBuildSteps }}
-
-          - ${{ if eq(job.customBuildSteps[0], '') }}:
             
-            - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
-              - task: NuGetToolInstaller@0
-                inputs:
-                  versionSpec: '4.9.2'
-                condition: eq(variables['_restoreInternalTools'], 'true')
-              - task: NuGetCommand@2
-                displayName: Restore internal tools
-                inputs:
-                  command: restore
-                  feedsToUse: config
-                  restoreSolution: 'eng\common\internal\Tools.csproj'
-                  nugetConfigPath: 'NuGet.config'
-                  restoreDirectory: '$(Build.SourcesDirectory)\.packages'
-                condition: eq(variables['_restoreInternalTools'], 'true')
-                env:
-                  VSS_NUGET_EXTERNAL_FEED_ENDPOINTS: '{"endpointCredentials": [{"endpoint":"https://devdiv.pkgs.visualstudio.com/_packaging/8f470c7e-ac49-4afe-a6ee-cf784e438b93/nuget/v3/index.json", "username":"dn-bot","password":"$(dn-bot-devdiv-nuget-feed-read)"}]}'
+          - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
+            - task: NuGetToolInstaller@0
+              inputs:
+                versionSpec: '4.9.2'
+              condition: ne(variables['_skipRestoreInternalTools'], 'true')
+            - task: NuGetCommand@2
+              displayName: Restore internal tools
+              inputs:
+                command: restore
+                feedsToUse: config
+                restoreSolution: 'eng\common\internal\Tools.csproj'
+                nugetConfigPath: 'NuGet.config'
+                restoreDirectory: '$(Build.SourcesDirectory)\.packages'
+              env:
+                VSS_NUGET_EXTERNAL_FEED_ENDPOINTS: '{"endpointCredentials": [{"endpoint":"https://devdiv.pkgs.visualstudio.com/_packaging/8f470c7e-ac49-4afe-a6ee-cf784e438b93/nuget/v3/index.json", "username":"dn-bot","password":"$(dn-bot-devdiv-nuget-feed-read)"}]}'
+                TargetGroup: $(_framework)
```

Comment:
There is no way to pass in extra msbuild properties to restore command. So I needed to set it as an env variable. I just checked with an official build and it works great!

Reply:
You have to invoke the target to get properties flowing.  EG: dotnet msbuild /t:Restore /p:Prop=value.

Reply:
That said, if the env var works then that's fine.

Reply:
Existing, but why do you need this environment variable?  Roslyn didn't use it: https://github.com/dotnet/roslyn/blob/971f984a3c1dc1ba156ef2cb44dd170df1fd8227/azure-pipelines-official.yml#L31-L38

Reply:
Because roslyn isn't restoring from a feed outside of dnceng which is outside of trust.

Reply:
I see.  We should fix that.  What's preventing ILC folks from pushing to dnceng?

Reply:
I agree. We have an internal thread discussing this but haven't yet found a consent.

Reply:
There is an offline thread on that going on.